hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -118,9 +158,196 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
+        requireNonNull(context);
         final var op = context.body().tokenRejectOrThrow();
+        final var rejectingAccountID = op.ownerOrElse(context.payer());
         final var rejections = op.rejections();
-        // Todo: Implement the logic for token rejection
+
+        final var hederaConfig = context.configuration().getConfigData(HederaConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        validateTrue(tokensConfig.tokenRejectEnabled(), NOT_SUPPORTED);
```

Comment:
I don't think this property is needed. Since this is a new transaction. @tinker-michaelj  What do you think ?

Reply:
The docs specified that we would need a feature toggle for TokenReject. - [Reference here](https://github.com/hashgraph/hedera-services/pull/12786/files#diff-345e91521530c2849705b8d557dc66a644189322a8cc02910e1b5ea11cd73929R24)

Reply:
I think it may be a specific for the HIP-904, as the other features have feature flag too. 

@MiroslavGatsanoga could probably answer this question better than me.

Reply:
@Neeharika-Sompalli I think the general idea for having feature flag for new features is so that they can be reverted quickly if needed. In the long term we should be cleaning up feature flags for features that don't need that capability e.g. feature has been live for X months. (we plan to do such clean up for HIP-904 afaik, cc @Nana-EC).

Reply:
Yes @MiroslavGatsanoga . Doesn't hurt. But it can be disabled easily by turning off throttles or permissions.